mhs-reference-implementation/mhs/routing/sds.py

```diff
@@ -27,25 +23,17 @@
 
 class SDSClient:
 
-    def __init__(self, ldap_address: str = None, tls: bool = False):
+    def __init__(self, sds_connection: ldap3.Connection, timeout: int = 3):
         """
-        :param ldap_address: The IP address of the directory containing data of the the MHSs
-        :param tls: Boolean defining whether or not to use TLS, this will require SSL certs to the LDAP server
+        :param sds_connection: takes an ldap connection to the sds server
         """
-        load_tls = None
-        if ldap_address:
-            if tls:
-                load_tls = ldap3.Tls(local_private_key_file=PRIVATE_KEY,
-                                     local_certificate_file=LOCAL_CERT_FILE,
-                                     validate=ssl.CERT_REQUIRED, version=ssl.PROTOCOL_TLSv1,
-                                     ca_certs_file=CA_CERTS_FILE)
-
-            server = ldap3.Server(ldap_address, use_ssl=tls, tls=load_tls)
-            self.connection = ldap3.Connection(server, auto_bind=True, client_strategy=ldap3.REUSABLE)
-        else:
-            self.connection = None
-
-    async def get_mhs_details(self, ods_code: str, interaction_id: str):
+        if not sds_connection:
+            raise ValueError('sds_connection must not be null')
+
+        self.connection = sds_connection
+        self.timeout = timeout
+
+    async def get_mhs_details(self, ods_code: str, interaction_id: str) -> Dict:
```

Comment:
Yep, I'd noticed this when looking at type hints, I refrained from using it here since the value type can be more than 1 type but I think in the future we should probably include the key/value types

Reply:
In that case, you could use `Any`, or `Union[someType, someOtherType]`, though that could get long-winded.